lib/trip_dispatcher.rb

```diff
@@ -90,6 +91,56 @@ def load_trips
       trips
     end
 
+    def find_last_trips
+      last_trips = []
+      drivers.each do |driver|
+        unless driver.trips.empty? || driver.trips.first.end_time == nil
+          last_trips << driver.trips.max_by {|trip| trip.end_time.to_i}
+        end
+      end
+      return last_trips
+    end
+
+    def least_recent
+
+      completed_trips = find_last_trips.find_all do |trip|
+        trip.end_time != nil && trip.driver.status == :AVAILABLE
+      end
```

Comment:
I don't think this will work if, somehow, the driver had an incomplete trip in the middle of their trip list. I only see your code checking the first and last trips.